api/models/prometheus.go

```diff
@@ -0,0 +1,101 @@
+package models
+
+import (
+	"sort"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/storage"
+)
+
+type PrometheusQueryRange struct {
+	Query   string `form:"query"`   //<string>: Prometheus expression query string.
+	Start   string `form:"start"`   //<rfc3339 | unix_timestamp>: Start timestamp.
+	End     string `form:"end"`     //<rfc3339 | unix_timestamp>: End timestamp.
+	Step    string `form:"step"`    //<duration>: Query resolution step width.
+	Timeout string `form:"timeout"` //<duration>: Evaluation timeout. Optional. Defaults to and is capped by the value of the -query.timeout flag.
+}
+
+type ProemtheusQueryData struct {
+	ResultType promql.ValueType `json:"resultType"`
+	Result     promql.Value     `json:"result"`
+}
+
+func NewPrometheusSeriesSet(series []storage.Series) *PrometheusSeriesSet {
+	return &PrometheusSeriesSet{
+		series: series,
+		cur:    0,
+	}
+}
+
+type PrometheusSeriesSet struct {
+	cur    int
+	series []storage.Series
+}
+
+func (p *PrometheusSeriesSet) Next() bool {
+	p.cur++
+	return p.cur-1 < len(p.series)
+}
+
+func (p *PrometheusSeriesSet) At() storage.Series {
+	return p.series[p.cur-1]
```

Comment:
what exactly does 'cur' mean if we have to return the element at cur minus one ? seems like 'cur' represents 'next'
or `At` means `Previous`.

Reply:
cur is the index of the element in absolute terms (starting from 1), for comparison with the length of the array.